charts/tidb-cluster/values.yaml

```diff
@@ -55,10 +55,35 @@ discovery:
 # if the ConfigMap was not changed.
 enableConfigMapRollout: true
 
-# Whether enable TLS connections between server nodes.
-# When enabled, PD/TiDB/TiKV/PUMP will use TLS encrypted connections to transfer data between each node,
-# certificates will be generated automatically (if not already present).
-enableTLSCluster: false
+# Whether enable the TLS connection between TiDB server components
+tlsCluster:
+  # When enabled, PD/TiDB/TiKV/PUMP will use TLS encrypted connections to transfer data between each node
+  enabled: false
+  # # secretName is the name of the secret that stores user-defined tidb components certificates, keys and ca
+  # # If not specified but tls cluster is enabled, certificates are created and signed automatically
+  # # To use user-defined certificates:
+  # #   1. First, generate TiDB server components certificates: https://pingcap.com/docs/stable/how-to/secure/generate-self-signed-certificates
+  # #   2. Then create the secret with the following command:
+  # #   kubectl create secret generic <secret-name> --namespace=<namespace>
+  # #       --from-file=pd-cert=<pd server certificate file path>
+  # #       --from-file=pd-key=<pd server key file path>
+  # #       --from-file=tikv-cert=<tikv server certificate file path>
+  # #       --from-file=tikv-key=<tikv server key file path>
+  # #       --from-file=tidb-cert=<tidb server certificate file path>
+  # #       --from-file=tidb-key=<tidb server key file path>
+  # #       --from-file=ca=<ca file path>
+  # secretName: "demo-tidb-components-secret"
```

Comment:
- one secret for one component
- for serects, I suggest using dedicated tls secret generation method in Kubernetes, `kubectl create secret tls <secret>`

the reasons are:
- automation tool like cert-manager manages one secret in each Certificate CRD 
- we should following k8s tls secret format, then third party generated secrets can be used directly

Reply:
We are considering using one certificate for all tidb components. If we can, just need one `secret`.

Reply:
Or TiDB core Components(include pd tikv and tidb) use one single cert, and TiDB Binlog(include pump and drainer) use another cert. It’s the same for future TiFlash and other components...

Reply:
@cofyc The `kubectl create secret tls` only have `--cert` and `--key`, but without `--ca` flag, so we can't use it to create the user-defined certificate secret?

Reply:
yes, we still need to use the `generic` type in this case.